lms/djangoapps/courseware/tabs.py

```diff
@@ -3,44 +3,347 @@
+    title = _('Courseware')
+    priority = 10
+    view_name = 'courseware'
+    is_movable = False
+
+
+class CourseInfoViewType(CourseViewType):
+    """
+    The course info view.
+    """
+    name = 'course_info'
+    title = _('Course Info')
+    priority = 20
+    view_name = 'info'
+    tab_id = 'info'
+    is_movable = False
+
+    @classmethod
+    def is_enabled(cls, course, user=None):
+        return True
 
 
-def get_course_tab_list(course, user):
+class SyllabusCourseViewType(EnrolledCourseViewType):
+    """
+    A tab for the course syllabus.
+    """
+    name = 'syllabus'
+    title = _('Syllabus')
+    priority = 30
+    view_name = 'syllabus'
+
+    @classmethod
+    def is_enabled(cls, course, user=None):  # pylint: disable=unused-argument
+        if not super(SyllabusCourseViewType, cls).is_enabled(course, user=user):
+            return False
+        return hasattr(course, 'syllabus_present') and course.syllabus_present
+
+
+class ProgressCourseViewType(EnrolledCourseViewType):
+    """
+    The course progress view.
+    """
+    name = 'progress'
+    title = _('Progress')
+    priority = 40
+    view_name = 'progress'
+    is_hideable = True
+
+    @classmethod
+    def is_enabled(cls, course, user=None):  # pylint: disable=unused-argument
+        if not super(ProgressCourseViewType, cls).is_enabled(course, user=user):
+            return False
+        return not course.hide_progress_tab
+
+
+class TextbookCourseViewsBase(CourseViewType):
+    """
+    Abstract class for textbook collection tabs classes.
+    """
+    # Translators: 'Textbooks' refers to the tab in the course that leads to the course' textbooks
+    title = _("Textbooks")
+    is_collection = True
+
+    @classmethod
+    def is_enabled(cls, course, user=None):  # pylint: disable=unused-argument
+        return user is None or user.is_authenticated()
+
+    @classmethod
+    def items(cls, course):
+        """
+        A generator for iterating through all the SingleTextbookTab book objects associated with this
+        collection of textbooks.
+        """
+        raise NotImplementedError()
+
+
+class TextbookCourseViews(TextbookCourseViewsBase):
+    """
+    A tab representing the collection of all textbook tabs.
+    """
+    name = 'textbooks'
+    priority = None
+
+    @classmethod
+    def is_enabled(cls, course, user=None):  # pylint: disable=unused-argument
+        parent_is_enabled = super(TextbookCourseViews, cls).is_enabled(course, user)
+        return settings.FEATURES.get('ENABLE_TEXTBOOK') and parent_is_enabled
+
+    @classmethod
+    def items(cls, course):
+        for index, textbook in enumerate(course.textbooks):
+            yield SingleTextbookTab(
+                name=textbook.title,
+                tab_id='textbook/{0}'.format(index),
+                link_func=lambda course, reverse_func, index=index: reverse_func(
+                    'book', args=[course.id.to_deprecated_string(), index]
+                ),
```

Comment:
I don't think I need to subtype `SingleTextbookTab`. I can probably just write an init method that refactors out the common stuff from the lambdas so that it's more readable.